common/src/main/java/org/apache/nemo/common/ir/vertex/utility/runtimepass/SignalVertex.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.nemo.common.ir.vertex.utility.runtimepass;
+
+import org.apache.nemo.common.ir.vertex.OperatorVertex;
+import org.apache.nemo.common.ir.vertex.executionproperty.MessageIdVertexProperty;
+import org.apache.nemo.common.ir.vertex.executionproperty.ParallelismProperty;
+import org.apache.nemo.common.ir.vertex.transform.SignalTransform;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Signal vertex holding signal transform.
+ */
+public final class SignalVertex extends OperatorVertex {
```

Comment:
Please provide tests for this new utility vertex

https://github.com/apache/incubator-nemo/blob/master/common/src/test/java/org/apache/nemo/common/ir/IRDAGTest.java#L313

Reply:
I've added some test methods. Thanks.